data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java

```diff
@@ -125,6 +166,42 @@ public void testEqualityDeletes() throws IOException {
     Assert.assertEquals("Table should contain expected rows", expected, actual);
   }
 
+  @Test
+  public void testEqualityDateDeletes() throws IOException {
+    initTable2();
+
+    Schema deleteRowSchema = table2.schema().select("*");
+    Record dataDelete = GenericRecord.create(deleteRowSchema);
+    List<Record> dataDeletes = Lists.newArrayList(
+            dataDelete.copy("dt", LocalDate.parse("2021-09-01"), "data", "a", "id", 1),
+            dataDelete.copy("dt", LocalDate.parse("2021-09-02"), "data", "b", "id", 2),
+            dataDelete.copy("dt", LocalDate.parse("2021-09-03"), "data", "c", "id", 3)
+    );
+
+    DeleteFile eqDeletes = FileHelpers.writeDeleteFile(
+            table2, Files.localOutput(temp.newFile()), Row.of(0), dataDeletes, deleteRowSchema);
```

Comment:
I don't see how this delete file would be applied. The partition you're writing it to is `Row.of(0)`, which ends up being `1970-01-01`. That should prevent the delete file from being found when planning, so none of the deletes would be applied.

Reply:
@xloya, what was the resolution to this? Were the deletes not applied? Was the test incorrect and not working?

Reply:
@rdblue Previously, the data file was written using Row.of(0), which means that all data is written to the 1970-01-01 partition. So all the equal delete data before is also written to Row.of(0). Now it is changed that the data is written to the corresponding date time partition, and then the equality delete data is also written to the corresponding partition.

Reply:
Makes sense. Thanks!